presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -1654,17 +1743,23 @@ public void updateNClob(String columnLabel, Reader reader)
         }
         Object object = getObject(columnIndex);
         if (object == null || type.isInstance(object)) {
-            //noinspection unchecked
-            return (T) object;
+            return type.cast(object);
+        }
+        try {
+            T converted = TYPE_CONVERSIONS.convert(object, type);
+            verify(converted != null, "Conversion cannot return null for non-null input, as this breask wasNull()");
+            return converted;
+        }
+        catch (NoConversionRegisteredException e) {
+            throw new SQLException(format("Cannot convert an instance of %s to %s", object.getClass(), type));
         }
-        throw new SQLException(format("Cannot convert an instance of %s to %s", object.getClass(), type));
     }
 
     @Override
     public <T> T getObject(String columnLabel, Class<T> type)
             throws SQLException
     {
-        throw new SQLFeatureNotSupportedException("getObject");
```

Comment:
extract commit?

Reply:
i didn't separate it because i did not see a clean separation, as the getObject(Class) is pointless without other changes. i can still try to do it, if you want